apollo-router/src/error.rs

```diff
@@ -549,26 +557,15 @@ impl std::fmt::Display for ValidationErrors {
 }
 
 impl ValidationErrors {
-    #[allow(clippy::needless_return)]
     pub(crate) fn print(&self) {
         if LevelFilter::current() == LevelFilter::OFF && cfg!(not(debug_assertions)) {
             return;
-        } else if atty::is(atty::Stream::Stdout) {
-            // Fancy reports for TTYs
-            self.errors.iter().for_each(|err| {
-                // `format!` works around https://github.com/rust-lang/rust/issues/107118
-                // to test the panic from https://github.com/apollographql/router/issues/2269
-                #[allow(clippy::format_in_format_args)]
-                {
-                    println!("{}", format!("{err}"));
-                }
-            });
-        } else {
-            // Best effort to display errors
-            self.errors.iter().for_each(|diag| {
-                println!("{}", diag.data);
-            });
-        };
+        }
+
+        self.errors.iter().for_each(|err| {
+            // Outputs a pretty colourised report on TTYs
+            eprintln!("{err}");
```

Comment:
We should really gate this entire function to test scope. We should never call println or variants in prod code. Otherwise if this is something that should  be used then it should use tracing.

Reply:
Makes sense! I've removed this function as it is only used in one place (supergraph schema validation error) and that already bubbles up the error. The bubbling up lost the actual underlying error message as described in #3681, so I've changed that in 525f376e4d4f288a1750742b427e736ccb88554b.